added options for a CA bundle

scottfp opened the pull request:
Feedback welcome. Thanks for the tool 🙂

krylovsk commented on main.go:
```diff
@@ -73,6 +75,7 @@ func main() {
 		clientPrefix = flag.String("client-prefix", "mqtt-benchmark", "MQTT client id prefix (suffixed with '-<client-num>'")
 		clientCert   = flag.String("client-cert", "", "Path to client certificate in PEM format")
 		clientKey    = flag.String("client-key", "", "Path to private clientKey in PEM format")
+		clientCaCert = flag.String("client-cacert", "", "Path to client CA certificate in PEM format")
```
Unless I'm missing something, this looks like you want to connect to a server with a TLS cert that is signed by a CA not known to the OS where the tool is run (e.g. self-signed or your private PKI)? 

Sounds to me that the proper name for this parameter would be `serverCACert` ?

scottfp replied:
You're right in that it's for when the CA is not known to the OS, and that this definitely isn't a client CA. I'm not sure I would consider it a broker CA, or just a root CA for both broker/client. Keeping it consistent seems wise though, so I pushed that change up

For what it's worth, here's what Mosquitto uses:
```
--cafile
    Define the path to a file containing PEM encoded CA certificates that are trusted. Used to enable SSL communication.

--cert
    Define the path to a file containing a PEM encoded certificate for this client, if required by the server.

--key
    Define the path to a file containing a PEM encoded private key for this client, if required by the server.
```
https://mosquitto.org/man/mosquitto_pub-1.html

krylovsk replied:
Yeah I see your point and now understand why it was initially 'client-ca', they both make sense we just think of it from different ends of the wire :) 

Sometimes it is indeed called just "CA", and that makes sense when both client and server use the same PKI. However, I still believe "broker CA" makes more sense since this tool is a client and it authenticates the broker and needs to know the CA that signed the broker's cert. Thanks!

krylovsk reviewed (changes requested):
Thanks for the contribution, this looks great sans the name of the parameter ;) 

Also, could you please update the README with the new `-h` output ? Thanks!

scottfp commented:
Two more things:
1) I removed the error when using broker CA without a client cert/key. This is a legitimate use case when connecting over SSL, but without cert-based authentication.
2) I added a flag for toggling TLS verify. It's defaulting to false now, with the flag enabling it. Please let me know if you want to keep it as it currently is, or if you feel this should be a separate PR.

krylovsk commented on main.go:
```diff
@@ -212,17 +216,28 @@ func printResults(results []*RunResults, totals *TotalResults, format string) {
+			log.Fatalf("Error reading CA certificate file: %v", err)
+		}
+		caCertPool = x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
```
Add a check that the certificate was loaded successfully ([AppendCertsFromPEM()]( https://pkg.go.dev/crypto/x509#CertPool.AppendCertsFromPEM) would return false if not)?

krylovsk reviewed (approved):
This looks great, agree with the points above too. Check that comment on validating that the read file actually contained a valid cert, otherwise looks good to me.

krylovsk commented:
@aparcar merged

aparcar commented:
That was quick, thanks.

krylovsk commented:
please pull latest `main`, I've fixed the comment above and renamed the flag to `-broker-ca-cert`.
